R/PipeOpFlatFun.R

```diff
@@ -60,14 +71,26 @@ PipeOpFlatFun = R6Class("PipeOpFlatFun",
         return(task)
       }
       pars = self$param_set$get_values()
-      interpolate = pars$interpolate
+      grid = pars$grid
 
       dt = task$data(cols = cols)
 
       flattened = imap(
         dt,
         function(x, nm) {
-          flat = suppressWarnings(as.matrix(x, interpolate = interpolate))
+          if (!is.character(grid)) {
+            flat = as.matrix(x, arg = grid, interpolate = TRUE)
+          } else if (grid == "union") {
+            flat = as.matrix(x, interpolate = TRUE)
+          } else {
+            args = tf::tf_arg(x)
```

Comment:
I think we should distinguish between reg and irreg case here (reg should be much easier)